File: java/src/test/java/SingletonTest.java
import org.junit.jupiter.api.Test;
import singleton.Singleton;

import static org.junit.jupiter.api.Assertions.assertSame;

public class SingletonTest {
    @Test
    void singletonTest() {
        final var instanceA = Singleton.getInstance();
        final var instanceB = Singleton.getInstance();

        assertSame(instanceA, instanceB);
    }
}

File: src/test/kotlin/SingletonTest.kt
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import io.kotest.matchers.types.shouldBeSameInstanceAs
import singleton.Singleton

class SingletonTest : FunSpec({
    test("Singleton Run") {
        val instanceA = Singleton
        val instanceB = Singleton

        instanceA shouldBeSameInstanceAs instanceB

        Singleton.function() shouldBe Singleton.field
    }
})